### PR TITLE
texlive: 2022-final -> 2023.202411

### PR DESCRIPTION
**Pull request by xworld21**
## Description of changes
Update TeX Live. The non-trivial changes are:
- the binaries for ConTeXt must be built separately
- dvisvgm now depends on mupdf for PDF handling
- the ConTeXt file database is now generated by a postaction script
- the ConTeXt file database is now reproducible in the correct way (my previous code would create multiple copies of the same UUIDs)

Since I was there, I removed the `texlive-` prefix from all package names. It's quite a rebuild so we should do this now.

This is based on #222142 so I added @apfelkuchen6 as coauthor – I hope that's ok!

## Things done
- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Closes: #222142

### CODE REVIEW

File: pkgs/tools/typesetting/tex/texlive/default.nix [hunk 1]
```diff
@@ -67,12 +67,13 @@ let
     # used for non-final snapshots and as fallback for final snapshots that have not reached yet the historic mirrors
     # please note that this server is not meant for large scale deployment and should be avoided on release branches
     # https://tug.org/pipermail/tex-live/2019-November/044456.html
+    "https://mirror.ox.ac.uk/sites/ctan.org/systems/texlive/tlnet"
```
Comment (xworld21) [inline pkgs/tools/typesetting/tex/texlive/default.nix:70]: Oops, that won't build starting from today (CTAN is surely distributing a new `tlpdb.xz` already!).

File: pkgs/tools/typesetting/tex/texlive/bin.nix [hunk 2]
```diff
@@ -309,46 +299,77 @@ chktex = stdenv.mkDerivation {
...
+  '';
+
   preConfigure = "cd texk/dvisvgm";
 
   configureFlags = common.configureFlags
     ++ [ "--with-system-kpathsea" ];
 
   nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ core brotli ghostscript zlib freetype woff2 potrace xxHash ];
+  buildInputs = [ core brotli ghostscript zlib freetype woff2 potrace xxHash mupdf ];
```
Comment (veprbl) [inline pkgs/tools/typesetting/tex/texlive/bin.nix:364]:
```suggestion
  buildInputs = [ core brotli ghostscript zlib freetype woff2 potrace xxHash mupdf-headless ];
```
#280285

Comment (xworld21) [reply pkgs/tools/typesetting/tex/texlive/bin.nix:364]: thanks, I didn't notice!

File: pkgs/tools/typesetting/tex/texlive/default.nix [hunk 3]
```diff
@@ -167,9 +174,8 @@ let
     scheme-minimal = [ free gpl1Only gpl2 gpl2Plus knuth lgpl21 lppl1 lppl13c mit ofl publicDomain ];
     scheme-small = [ asl20 cc-by-40 cc-by-sa-40 cc0 fdl13Only free gfl gpl1Only gpl2 gpl2Plus gpl3 gpl3Plus knuth
       lgpl2 lgpl21 lppl1 lppl12 lppl13a lppl13c mit ofl publicDomain x11 ];
-    scheme-tetex = [ artistic1-cl8 asl20 bsd2 bsd3 cc-by-40 cc-by-sa-10 cc-by-sa-20 cc-by-sa-30 cc-by-sa-40 cc0
-      fdl13Only free gfl gpl1Only gpl2 gpl2Plus gpl3 gpl3Plus isc knuth lgpl2 lgpl21 lgpl3 lppl1 lppl12 lppl13a
-      lppl13c mit ofl publicDomain x11];
+    scheme-tetex = [ artistic1-cl8 asl20 bsd2 bsd3 cc-by-30 cc-by-40 cc-by-sa-10 cc-by-sa-20 cc-by-sa-30 cc-by-sa-40 cc0 fdl13Only free gfl gpl1Only
```
[the remark this answers is not in the document]
Comment (xworld21) [reply pkgs/tools/typesetting/tex/texlive/default.nix:177]: The correct list is printed out by `tests.texlive.licenses`.

File: pkgs/tools/typesetting/tex/texlive/bin.nix (hunk 2 above)
Comment (veprbl) [reply pkgs/tools/typesetting/tex/texlive/bin.nix]: Actually we need to revert to mupdf for now. We can follow up on staging to switch to headless.

Comment (xworld21) [reply pkgs/tools/typesetting/tex/texlive/bin.nix:364]: I see. Reverted to `mupdf` now.

Comment (SuperSandro2000) [reply pkgs/tools/typesetting/tex/texlive/bin.nix:364]: Should we add a TODO or maybe send a PR to staging after this got merged?

Comment (dotlambda) [reply pkgs/tools/typesetting/tex/texlive/bin.nix:364]: The latter.

**Comment by gepbird** [comment]
Result of `nixpkgs-review pr 281054` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>23 packages marked as broken and skipped:</summary>
  <ul>
    <li>golden-cheetah-bin</li>
    <li>python312Packages.abjad</li>
    <li>python312Packages.abjad.dist</li>
    <li>python312Packages.nbdev</li>
    <li>python312Packages.nbdev.dist</li>
    <li>python312Packages.pplpy</li>
    <li>python312Packages.pplpy.dist</li>
    <li>python312Packages.pplpy.doc</li>
    <li>python312Packages.pytikz-allefeld</li>
    <li>python312Packages.pytikz-allefeld.dist</li>
    <li>python312Packages.recoll</li>
    <li>python312Packages.rpy2</li>
    <li>python312Packages.rpy2.dist</li>
    <li>python312Packages.sphinxcontrib-bayesnet</li>
    <li>python312Packages.sphinxcontrib-bayesnet.dist</li>
    <li>python312Packages.sphinxcontrib-tikz</li>
    <li>python312Packages.sphinxcontrib-tikz.dist</li>
    <li>rocmPackages.rdc</li>
    <li>rocmPackages.rdc.doc</li>
    <li>rocmPackages_5.rdc</li>
    <li>rocmPackages_5.rdc.doc</li>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>29 packages failed to build:</summary>
  <ul>
    <li>auctex</li>
    <li>auctex.tex</li>
    <li>blahtexml</li>
    <li>blahtexml.doc</li>
    <li>cantor (libsForQt5.cantor ,plasma5Packages.cantor)</li>
    <li>frescobaldi</li>
    <li>frescobaldi.dist</li>
    <li>labplot</li>
    <li>pynac</li>
    <li>python311Packages.abjad</li>
    <li>python311Packages.abjad.dist</li>
    <li>python311Packages.cnvkit</li>
    <li>python311Packages.cnvkit.dist</li>
    <li><s>python311Packages.radian</s> (built on second try)</li>
    <li>python311Packages.radian.dist</li>
    <li>python311Packages.unstructured</li>
    <li>python311Packages.unstructured.dist</li>
    <li>python312Packages.cypari2</li>
    <li>python312Packages.cypari2.dist</li>
    <li>rocmPackages.mivisionx (rocmPackages.mivisionx-hip ,rocmPackages_5.mivisionx ,rocmPackages_5.mivisionx-hip)</li>
    <li>rocmPackages.mivisionx-cpu (rocmPackages_5.mivisionx-cpu)</li>
    <li>rocmPackages.mivisionx-opencl (rocmPackages_5.mivisionx-opencl)</li>
    <li>rocmPackages.rocprofiler (rocmPackages_5.rocprofiler)</li>
    <li>sage</li>
    <li>sageWithDoc</li>
    <li>xyce</li>
    <li><s>xyce-parallel</s> (built on second try)</li>
    <li>xyce-parallel.doc</li>
    <li>xyce.doc</li>
  </ul>
</details>
<details>
  <summary>222 packages built:</summary>
  <ul>
    <li>R</li>
    <li>R.tex</li>
    <li>advi</li>
    <li>apostrophe</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>asciidoc-full-with-plugins.dist</li>
    <li>asciidoc-full.dist</li>
    <li>asymptote</li>
    <li>auto-multiple-choice</li>
    <li>avrdude</li>
    <li>avrdudess</li>
    <li>biber</li>
    <li>biber-ms</li>
    <li>biber-ms.devdoc</li>
    <li>biber.devdoc</li>
    <li>bluespec</li>
    <li>bluespec.doc</li>
    <li>catdvi</li>
    <li>catdvi.dev</li>
    <li>cddlib</li>
    <li>clevis</li>
    <li>clevis.man</li>
    <li>dblatex</li>
    <li>dblatexFull</li>
    <li>denemo</li>
    <li>diffoscope</li>
    <li>diffoscope.dist</li>
    <li>diffoscope.man</li>
    <li>digikam</li>
    <li>disorderfs</li>
    <li>dot2tex (python311Packages.dot2tex)</li>
    <li>dot2tex.dist (python311Packages.dot2tex.dist)</li>
    <li>dwarf-fortress-packages.dwarf-fortress-full</li>
    <li>dwarf-therapist (dwarf-fortress-packages.dwarf-therapist)</li>
    <li>dwarf-fortress-packages.dwarf-therapist-original</li>
    <li>eclib</li>
    <li>enblend-enfuse</li>
    <li>evince (gnome.evince)</li>
    <li>evince.dev (gnome.evince.dev)</li>
    <li>evince.devdoc (gnome.evince.devdoc)</li>
    <li>gap</li>
    <li>gap-full</li>
    <li>gap-minimal</li>
    <li>gfan</li>
    <li>giac</li>
    <li>giac-with-xcas</li>
    <li>giac.doc</li>
    <li>glom</li>
    <li>glom.dev</li>
    <li>glom.devdoc</li>
    <li>glom.doc</li>
    <li>glom.lib</li>
    <li>gnome.sushi</li>
    <li>gnu-cobol</li>
    <li>gnu-cobol.bin</li>
    <li>gnu-cobol.dev</li>
    <li>gnu-cobol.lib</li>
    <li>gp2c</li>
    <li>gummi</li>
    <li>herwig</li>
    <li>hqplayer-desktop</li>
    <li>hqplayer-desktop.doc</li>
    <li>hugin</li>
    <li>inkscape-extensions.textext</li>
    <li>inkscape-extensions.textext.dist</li>
    <li>ipe</li>
    <li>iwona</li>
    <li>jasp-desktop</li>
    <li>jetbrains.dataspell</li>
    <li>rep (kakounePlugins.rep)</li>
    <li>klipper-firmware</li>
    <li>klipper-flash</li>
    <li>latte-integrale</li>
    <li>lcalc</li>
    <li>lilypond</li>
    <li>lilypond-unstable</li>
    <li>lilypond-unstable-with-fonts</li>
    <li>lilypond-with-fonts</li>
    <li>luksmeta</li>
    <li>manga-cli</li>
    <li>manim</li>
    <li>manim.dist</li>
    <li>mate.atril</li>
    <li>mitscheme</li>
    <li>mitschemeX11</li>
    <li>ne</li>
    <li>nix-init</li>
    <li>nuweb</li>
    <li>openmolcas</li>
    <li>opentx</li>
    <li>paperwork</li>
    <li>paperwork.dist</li>
    <li>pari</li>
    <li>phosh</li>
    <li>phosh-mobile-settings</li>
    <li>polymake</li>
    <li>postgresql12JitPackages.plr</li>
    <li>postgresql12Packages.plr</li>
    <li>postgresql13JitPackages.plr</li>
    <li>postgresql13Packages.plr</li>
    <li>postgresql14JitPackages.plr</li>
    <li>postgresql14Packages.plr</li>
    <li>postgresqlJitPackages.plr (postgresql15JitPackages.plr)</li>
    <li>postgresqlPackages.plr (postgresql15Packages.plr)</li>
    <li>postgresql16JitPackages.plr</li>
    <li>postgresql16Packages.plr</li>
    <li>python311Packages.cypari2</li>
    <li>python311Packages.cypari2.dist</li>
    <li>python311Packages.cysignals</li>
    <li>python311Packages.cysignals.dist</li>
    <li>python311Packages.fpylll</li>
    <li>python311Packages.fpylll.dist</li>
    <li>python311Packages.nbdev</li>
    <li>python311Packages.nbdev.dist</li>
    <li>python311Packages.pplpy</li>
    <li>python311Packages.pplpy.dist</li>
    <li>python311Packages.pplpy.doc</li>
    <li>python311Packages.primecountpy</li>
    <li>python311Packages.primecountpy.dist</li>
    <li>python311Packages.pylatex</li>
    <li>python311Packages.pylatex.dist</li>
    <li>python311Packages.pypandoc</li>
    <li>python311Packages.pypandoc.dist</li>
    <li>python311Packages.pytikz-allefeld</li>
    <li>python311Packages.pytikz-allefeld.dist</li>
    <li>python311Packages.rchitect</li>
    <li>python311Packages.rchitect.dist</li>
    <li>recoll (python311Packages.recoll)</li>
    <li>rivet (python311Packages.rivet)</li>
    <li>python311Packages.rpy2</li>
    <li>python311Packages.rpy2.dist</li>
    <li>python311Packages.sphinxcontrib-bayesnet</li>
    <li>python311Packages.sphinxcontrib-bayesnet.dist</li>
    <li>python311Packages.sphinxcontrib-tikz</li>
    <li>python311Packages.sphinxcontrib-tikz.dist</li>
    <li>python311Packages.torchWithRocm</li>
    <li>python311Packages.torchWithRocm.cxxdev</li>
    <li>python311Packages.torchWithRocm.dev</li>
    <li>python311Packages.torchWithRocm.dist</li>
    <li>python311Packages.torchWithRocm.lib</li>
    <li>python312Packages.cysignals</li>
    <li>python312Packages.cysignals.dist</li>
    <li>python312Packages.dot2tex</li>
    <li>python312Packages.dot2tex.dist</li>
    <li>python312Packages.fpylll</li>
    <li>python312Packages.fpylll.dist</li>
    <li>python312Packages.primecountpy</li>
    <li>python312Packages.primecountpy.dist</li>
    <li>python312Packages.pylatex</li>
    <li>python312Packages.pylatex.dist</li>
    <li>python312Packages.pypandoc</li>
    <li>python312Packages.pypandoc.dist</li>
    <li>python312Packages.radian</li>
    <li>python312Packages.radian.dist</li>
    <li>python312Packages.rchitect</li>
    <li>python312Packages.rchitect.dist</li>
    <li>python312Packages.rivet</li>
    <li>qmk</li>
    <li>qmk.dist</li>
    <li>quarto</li>
    <li>rocmPackages.migraphx (rocmPackages_5.migraphx)</li>
    <li>rocmPackages.miopen (rocmPackages.miopen-hip ,rocmPackages_5.miopen ,rocmPackages_5.miopen-hip)</li>
    <li>rocmPackages.miopen-opencl (rocmPackages_5.miopen-opencl)</li>
    <li>rocmPackages.miopengemm (rocmPackages_5.miopengemm)</li>
    <li>rocmPackages.miopengemm.doc (rocmPackages_5.miopengemm.doc)</li>
    <li>rocmPackages.rocdbgapi (rocmPackages_5.rocdbgapi)</li>
    <li>rocmPackages.rocdbgapi.doc (rocmPackages_5.rocdbgapi.doc)</li>
    <li>rocmPackages.rocr-debug-agent (rocmPackages_5.rocr-debug-agent)</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>sagetex.tex</li>
    <li>sherpa</li>
    <li>singular</li>
    <li>skribilo</li>
    <li>solfege</li>
    <li>spark (spark3 ,spark_3_5)</li>
    <li>spark_3_3</li>
    <li>spark_3_4</li>
    <li>sympow</li>
    <li>tang</li>
    <li>tang.man</li>
    <li>tests.nixpkgs-check-by-name</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.libR</li>
    <li>texliveBasic</li>
    <li>texliveBasic.info (texliveBookPub.info ,texliveConTeXt.info ,texliveMinimal.info ,texliveSmall.info)</li>
    <li>texliveBasic.man</li>
    <li>texliveBookPub</li>
    <li>texliveBookPub.man</li>
    <li>texliveConTeXt</li>
    <li>texliveConTeXt.man</li>
    <li>texliveFull</li>
    <li>texliveFull.info</li>
    <li>texliveFull.man</li>
    <li>texliveGUST</li>
    <li>texliveGUST.info</li>
    <li>texliveGUST.man</li>
    <li>texliveInfraOnly</li>
    <li>texliveInfraOnly.info</li>
    <li>texliveInfraOnly.man</li>
    <li>texliveMedium</li>
    <li>texliveMedium.info</li>
    <li>texliveMedium.man</li>
    <li>texliveMinimal</li>
    <li>texliveMinimal.man</li>
    <li>texliveSmall</li>
    <li>texliveSmall.man</li>
    <li>texliveTeTeX</li>
    <li>texliveTeTeX.info</li>
    <li>texliveTeTeX.man</li>
    <li>texmacs</li>
    <li>thepeg</li>
    <li>therion</li>
    <li>tinygo</li>
    <li>vimPlugins.openscad-nvim</li>
    <li>wayback-machine-archiver</li>
    <li>wayback-machine-archiver.dist</li>
    <li>yosys-bluespec</li>
    <li>zathura</li>
    <li>zettlr</li>
  </ul>
</details>

Most of the failed packages are failing before this PR, but there are 2 related failures:
<details><summary>2 new failures</summary>
<details><summary>auctex: ! LaTeX Error: File `etoolbox.sty' not found.</summary>

```
(/nix/store/w2v6sxxhrxkvh631ln5n8c5fgki372vg-texlive-2023-env-texmfdist/tex/lat
ex/hyperref/nameref.sty
(/nix/store/w2v6sxxhrxkvh631ln5n8c5fgki372vg-texlive-2023-env-texmfdist/tex/lat
ex/refcount/refcount.sty)
(/nix/store/w2v6sxxhrxkvh631ln5n8c5fgki372vg-texlive-2023-env-texmfdist/tex/gen
eric/gettitlestring/gettitlestring.sty
(/nix/store/w2v6sxxhrxkvh631ln5n8c5fgki372vg-texlive-2023-env-texmfdist/tex/lat
ex/kvoptions/kvoptions.sty)))

! LaTeX Error: File `etoolbox.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 
! Emergency stop.
<read *> 
         
l.119 \def
          \Hy@Error{\PackageError{hyperref}}^^M
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on preview.log.
make[1]: *** [Makefile:91: preview.pdf] Error 1
make[1]: Leaving directory '/build/auctex-13.2/latex'
make: *** [Makefile:244: texmf] Error 2
```
</details>
<details><summary>blahtexml</summary>

```
(/nix/store/l5zf0aw83hlp5yil43126v7l1b1h6ncx-texlive-2023-env-texmfdist/tex/lat
ex/epstopdf-pkg/epstopdf-base.sty
(/nix/store/l5zf0aw83hlp5yil43126v7l1b1h6ncx-texlive-2023-env-texmfdist/tex/lat
ex/latexconfig/epstopdf-sys.cfg))
! Argument of <C2> has an extra }.
<inserted text> 
                \par 
l.55 \section{Introduction}
                           
? 
! Emergency stop.
<inserted text> 
                \par 
l.55 \section{Introduction}
                           
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on manual.log.
make: *** [makefile:139: Documentation/manual.pdf] Error 1
```
</details>
</details>

**Comment by veprbl** [comment]
I wasn't able to build this on either macOS or linux. This is some old nix regression, I believe
```
error: opening directory '/nix/store/3mbcjxm62v15nb4f8k1m390svb02hip1-quizztex.r68823.tar.xz': Too many open files
```

**Comment by olynch** [comment]
I got a different error:

```
> nix shell 'github:xworld21/nixpkgs/texlive-20240114#texliveSmall'                                                                                                                                     
error: derivation '/nix/store/bfh4bw2mqavfnizd32zbrs3dplcsn32s-texlive-2023-env.drv' does not have wanted outputs 'man'
```

**Comment by xworld21** [comment]
I fixed auctex and blahtexml, and the missing output issue should have disappeared after rebasing to master.
 
> ```
> error: opening directory '/nix/store/3mbcjxm62v15nb4f8k1m390svb02hip1-quizztex.r68823.tar.xz': Too many open files
> ```

Not sure about what to do for this one, though, that is a sandbox issue.

**Comment by veprbl** [comment]
> Not sure about what to do for this one, though, that is a sandbox issue.

Not a direct fault of texlive. I was able to build it locally in a different way.

**Review by veprbl — APPROVED** [review]
Everything seems to work

**Comment by veprbl** [comment]
@GrahamcOfBorg build tests.texlive

**Comment by veprbl** [comment]
@xworld21 Thank you!

**Pull request merged**